File: binary-tree-preorder-traversal/binary-tree-preorder-traversal.c
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     struct TreeNode *left;
 *     struct TreeNode *right;
 * };
 */


/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int nodecount(struct TreeNode *root){
    if (root == NULL){
        return 0;
    }
   
    return 1 + nodecount(root->left) + nodecount(root->right);
}
void insert(struct TreeNode* root, int* array, int* i){
    if (root == NULL){
        return;
    }
    
    
    array[(*i)] = root->val;
    (*i)++;
    insert(root->left, array, i);
    insert(root->right, array, i);
}
int* preorderTraversal(struct TreeNode* root, int* returnSize){
    *returnSize = nodecount(root);
    int *array = (int*)malloc(sizeof(int) * *returnSize);
    
    int* index = 0;
    insert(root, array, &index);
    
    return array;
    
    
}

File: 144-binary-tree-preorder-traversal/binary-tree-preorder-traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def preorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        stack, arr = [], []
        curr = root
        stack.append(root)
        
        while len(stack) != 0:
            node = stack.pop()
            if node:
                arr.append(node.val)

                if node.right:
                    stack.append(node.right)
                if node.left:
                    stack.append(node.left)
        
        
        
        return arr